### DISCUSSION TITLE
My Huggingface model gives an error when trying "Hosted Inference API"

### DISCUSSION THREAD

**Discussion by CanKeles5** [asteroid-team/asteroid#602] [Q&A] [answered]
I trained a ConvTasNet model on a custom dataset and I uploaded the model to HuggingFace. The model works fine when I try to use it by loading it from huggingface with "ConvTasNet.from_pretrained("/link/to/model")", but when I try the hosted inference api it gives me the following error: "No module named 'torch.torch_version'".

The model is available here: https://huggingface.co/cankeles/ConvTasNet_WHAMR_enhsingle_16k

I am wondering if I am doing something wrong.

**Comment by JorisCos** [comment, marked as answer]
Weird, I have checked several models uploaded by our community on HF some work some don't with an `unknown error` @mpariente is it from our side or HF's ?

**Reply by mpariente** [reply]
If the model works when imported `from_pretrained`, that's probably a problem on the server side, but not sure where to look. @julien-c do you know what might be the problem please?

**Reply by julien-c** [reply]
tagging @osanseviero @narsil!

**Reply by osanseviero** [reply]
Thanks for flagging! https://huggingface.co/JorisCos/ConvTasNet_Libri2Mix_sepclean_8k seems to be working, but https://huggingface.co/cankeles/ConvTasNet_WHAMR_enhsingle_16k does not. I tried loading with versions we use in the Inference API and I can load the model, so there seems to be some deeper issue here. This definitively seems like an issue from our side, sorry for it!

**Reply by mpariente** [reply]
Thanks for the quick reaction ! 
Could it be a problem of reading files/resampling related to sampling rate? (both non-working models are 16k)

**Reply by CanKeles5** [reply]
Hi, thank you all for the quick responses. I uploaded the models manually, could that cause a problem? Maybe the tags could be problematic too? I first used a different dataset tag that wasnt already in use and I changed it later to LibriMix.

**Reply by osanseviero** [reply]
Given the error is related to pytorch imports I don't think it's an issue with sampling rate. The tags don't have any impact in the Inference code. In case you're curious, the code is in https://github.com/huggingface/huggingface_hub/blob/main/api-inference-community/docker_images/asteroid/app/pipelines/audio_to_audio.py.

**Reply by Narsil** [reply]
This is now fixed. we merely rebuilt the docker file.

It's probably linked to the fact that your code runs on `torch==1.10` where the docker image was running an older version of torch (torch_version.py  seems new: https://github.com/pytorch/pytorch/blame/master/torch/torch_version.py, only 7 months old)

Since the model is serialized through pickle, it contained the torch_version in your file, and it couldn't unpickle it on an older lib.

Since asteroid does not define a hard number of torch version, the docker building took the latest version 1.10 and everything worked.

**Reply by mpariente** [reply]
Thanks everybody !
